### PR TITLE
Add more syntax tests for Yul

### PR DESCRIPTION
**Pull request by axic**
Pulled out of #9549. The reason to merge it here, because #9549 may not be merged until 0.8.0.

Some of these tests are missing entirely currently, like the octal one.

### CODE REVIEW

File: test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    function loadimmutable(a) {}
+}
+// ====
+// dialect: evm
+// ----
+// ParserError 5568: (15-28): Cannot use builtin function name "loadimmutable" as identifier name.
```
Comment (axic) [inline test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul:7]: As @ekpyron pointed out: perhaps this should not happen in the EVM dialect, only in the EVM object dialect.

However as it turned out, the `evm` dialect in yul syntax tests refers to evm with objects: https://github.com/ethereum/solidity/blob/develop/test/libyul/Common.cpp#L115-L117

This was changed in a5ae51fa6e (#8442) where during refactoring it was to enable objects. I'm not sure if it was planned or not, @a3d4 @chriseth ?

Comment (ekpyron) [reply test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul:7]: We need to find a way to have both: decent coverage for the inline assembly dialect (we only have very few actual solidity tests using inline assembly) and the object dialect - at best without duplicating or triplicating all of them... we probably need to have most tests run against multiple dialects and then a few ones running against only a subset or something like that...

Comment (ekpyron) [reply test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul:7]: Arguably, the builtin errors shouldn't be caught by the parser anyways, but rather as an analysis step... then most parser tests could be shared (that is except for dot-identifier-tests :-D)

**Review by ekpyron — APPROVED** [review]
The dialect thing has to be fixed... but apparently we're deciding to do that separately so given that we might as well merge this...

File: test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul (hunk 1 above)
Comment (a3d4) [reply test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul:7]: > This was changed in a5ae51f (#8442) where during refactoring it was to enable objects. I'm not sure if it was planned or not, @a3d4 @chriseth ?

Not sure anymore, but I think it is just a side effect of refactoring. Should I revert?

Comment (axic) [reply test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul:7]: No, we have a discussion issue under #9620 where we can sort this out later.

**Pull request merged**